x/ccv/provider/types/msg.go

```diff
@@ -388,6 +389,11 @@ func (msg MsgUpdateConsumer) ValidateBasic() error {
 		}
 	}
 
+	if msg.NewChainId != "" && len(msg.NewChainId) > cmttypes.MaxChainIDLen {
```

Comment:
Create a utility method to validate consumer chain IDs as it's needed for both create and update messages. Also, don't forget to check that the chainId is not neutron-1 or stride-1.

Reply:
Good catch in regards to Neutron and Stride. Thanks!

I created the `ValidateChainId` utility method.  Note however that it cannot be used uniformly across `MsgCreateConsumer` and `MsgUpdateConsumer`. The reason being that the `chainId` of `MsgCreateConsumer` _has to be_ non non empty, while `MsgUpdateConsumer`'s `newChainId` could be empty or it could be the exact same as the chain id the consumer chain currently has.

Because of the above, `ValidateChainId` is used in the `MsgCreateConsumer.ValidateBasic`, while `MsgUpdateConsumer.ValidateBasic` only checks the length of the `newChainId` and performs the `ValidateChainId` in `UpdateConsumer`.